File: frontend/src/services/auth.service.ts
import { apiPost, apiGet } from "./api";
import type { LoginResult, AuthUser } from "../types";

export interface LoginData {
  email: string;
  password: string;
}

export interface RegisterData {
  nombre: string;
  apellido: string;
  genero: "M" | "F";
  email: string;
  password: string;
  rol_id: number;
}

export const authService = {
  login: (data: LoginData) => apiPost<LoginResult>("/auth/login", data),
  register: (data: RegisterData) => apiPost<LoginResult>("/auth/register", data),
  profile: () => apiGet<AuthUser>("/auth/profile"),
};

File: flutter_signature/lib/features/auth/services/auth_service.dart
import 'package:onesignal_flutter/onesignal_flutter.dart';

import '../models/user_model.dart';
import '../../../core/api/dio_client.dart';
import '../../../core/api/endpoints.dart';
import '../../../core/storage/secure_storage.dart';

class AuthService {
  Future<UserModel?> login({
    required String email,
    required String password,
  }) async {
    await DioClient.ensureInitialized();

    final response = await DioClient.dio.post(
      Endpoints.login,
      data: {'email': email, 'password': password},
    );

    final body = response.data;
    if (body['ok'] != true || body['data'] == null) {
      return null;
    }

    final data = body['data'];
    final userData = data['user'] as Map<String, dynamic>;
    final token = data['token'] as String;

    final user = UserModel.fromJson(userData, token);

    await SecureStorage.saveToken(token);
    await SecureStorage.saveUserData(
      id: user.id,
      name: user.nombre,
      lastName: user.apellido,
      email: user.email,
      rolId: user.rolId,
    );

    await OneSignal.login(user.id.toString());

    return user;
  }

  Future<void> logout() async {
    await OneSignal.logout();
    await SecureStorage.clearAll();
  }
}
